flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxyV2Factory.java

```diff
@@ -55,25 +58,39 @@ public static KinesisProxyV2Interface createKinesisProxyV2(final Properties conf
         final AttributeMap.Builder clientConfiguration = AttributeMap.builder();
         populateDefaultValues(clientConfiguration);
 
-        final SdkAsyncHttpClient httpClient =
+        final SdkHttpClient httpClient =
+                AWSGeneralUtil.createSyncHttpClient(
+                        convertedProperties.merge(clientConfiguration.build()),
+                        ApacheHttpClient.builder());
+
+        final SdkAsyncHttpClient asyncHttpClient =
                 AWSGeneralUtil.createAsyncHttpClient(
                         convertedProperties.merge(clientConfiguration.build()),
                         NettyNioAsyncHttpClient.builder());
+
         final FanOutRecordPublisherConfiguration configuration =
                 new FanOutRecordPublisherConfiguration(configProps, emptyList());
 
-        Properties asyncClientProperties =
-                KinesisConfigUtil.getV2ConsumerAsyncClientProperties(configProps);
+        Properties clientProperties = KinesisConfigUtil.getV2ConsumerClientProperties(configProps);
 
-        final KinesisAsyncClient client =
+        final KinesisAsyncClient asyncClient =
                 AWSAsyncSinkUtil.createAwsAsyncClient(
-                        asyncClientProperties,
-                        httpClient,
+                        clientProperties,
+                        asyncHttpClient,
                         KinesisAsyncClient.builder(),
                         KinesisStreamsConfigConstants.BASE_KINESIS_USER_AGENT_PREFIX_FORMAT,
                         KinesisStreamsConfigConstants.KINESIS_CLIENT_USER_AGENT_PREFIX);
 
-        return new KinesisProxyV2(client, httpClient, configuration, BACKOFF);
+        final KinesisClient client =
+                AWSAsyncSinkUtil.createAwsSyncClient(
+                        clientProperties,
+                        httpClient,
+                        KinesisClient.builder(),
+                        KinesisStreamsConfigConstants.BASE_KINESIS_USER_AGENT_PREFIX_FORMAT,
+                        KinesisStreamsConfigConstants.KINESIS_CLIENT_USER_AGENT_PREFIX);
+
+        return new KinesisProxyV2(
+                asyncClient, client, asyncHttpClient, httpClient, configuration, BACKOFF);
```

Comment:
As we discussed, we should split `KinesisProxyV2` so we do not need to create both client. In the case of registerStreamConsumer it will not use the async client and it is expensive to create a thread pool, then trash it. We should only create the async client when we actually need to use it

Reply:
Thank you for clarifying, I have split `KinesisProxyV2` to `KinesisProxySyncV2` and `KinesisProxyAsyncV2`. 

For StreamRegistrar related operations (i.e registerStreamConsumer), only `KinesisProxySyncV2` will be created. 
For EFO related operations (i.e subsribeToShard), only `KinesisProxyAsyncV2` will be created.